mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -1342,6 +1359,10 @@ impl ObjectClient for S3CrtClient {
     }
 }
 
+pub trait OnTelemetry: std::fmt::Debug + Send + Sync {
+    fn on_telemetry(&self, request_metrics: &RequestMetrics, operation: &str);
```

Comment:
I agree, i left `operation` because the existing telemetry code using it extensively

```
                    metrics::counter!("s3.requests.failures", "op" => op, "type" => request_type, "status" => http_status.unwrap_or(-1).to_string()).increment(1);
```


If you think it's unnecessary, will remove it.